evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -58,6 +60,15 @@ def _prepare_data(self, X, y):
             pd.DataFrame, pd.Series: Prepared X and y data as pandas types
         """
         X = infer_feature_types(X)
+        try:
+            X = X.astype(
+                {null_col: int for null_col in X.ww.select(IntegerNullable).columns},
+            )
+        except ValueError:
+            X = X.astype(
+                {null_col: float for null_col in X.ww.select(IntegerNullable).columns},
+            )
+        X.ww.init()
```

Comment:
It seems rather redundant to call `X.ww.init()` so close after calling `infer_feature_types`. Can we do some of this type shifting directly through woodwork instead?

Reply:
Addressed this, passed through the schema...hopefully in a way that makes you proud.